bpaotu/bpaotu/static/bpaotu/bpaotu.js

```diff
@@ -108,11 +109,14 @@ $(document).ready(function() {
     };
 
     var taxonomy_refresh = function() {
+        check_for_auth_token();
```

Comment:
this isn't quite right - you should use jQuery's futures to chain together the request for auth (if needed), and then the actual `$.ajax` call. that `check_for_auth_token()` function as it stands will fire off a future (the AJAX request) but that almost certainly won't have completed by the time you move on and fire off the live request.

Reply:
Implemented.